testsuite/MDAnalysisTests/analysis/test_bat.py

```diff
@@ -127,3 +128,13 @@ def test_bat_incorrect_dims(self, bat_npz):
         errmsg = 'Dimensions of array in loaded file'
         with pytest.raises(ValueError, match=errmsg):
             R = BAT(selected_residues, filename=bat_npz)
+
+    def test_bat_transformation(self, selected_residues, bat):
+        R = BAT(selected_residues)
+        R.run()
+        XYZ = R.results.bat[0]
+        PQR = copy.deepcopy(XYZ)
+        R.Cartesian(R.results.bat[0])
+        assert_almost_equal(PQR, XYZ, 5,
+                            err_msg="error: Bat input data"
+                                    "modified by the Cartesian method")
```

Comment:
How about this instead? The below suggestion does not need to `.run()` again, which saves time. It also has more informative function and variable nams.

```suggestion
    def test_Cartesian_does_not_modify_input(self, selected_residues, bat):
        R = BAT(selected_residues)
        pre_transformation = copy.deepcopy(bat[0])
        R.Cartesian(bat[0])
        assert_almost_equal(
            pre_transformation, bat[0],
            err_msg="BAT.Cartesian modified input data"
        )
```